File: 0513/513.py
class Solution:
    def find_bottom_left_tree_value(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.find_bottom_left_tree_value(root.left), self.find_bottom_left_tree_value(root.right))

File: 0513/513.java

class Solution {
    public int findBottomLeftValue(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        int res = root.val;
        while (!queue.isEmpty()) {
            res = queue.peek().val;
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode node = queue.poll();
                if (node.left != null) queue.offer(node.left);
                if (node.right != null) queue.offer(node.right);
            }
        }
        return res;
    }
}
